vermouth/rcsu/get_go.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2023 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Process the Go contact pairs.
+"""
+import numpy as np
+import networkx as nx
+from vermouth.molecule import Interaction
+
+def _get_bb_nodes(molecule, nodes):
+    for node in nodes:
+        if molecule.nodes[node]['atomname'] == "BB":
+            return node
+    return None
+
+def _get_go_type(molecule, resid, chain, prefix):
+    for node in molecule.nodes:
+        attrs = molecule.nodes[node]
+        if attrs['resid'] == resid and attrs['chain'] == chain:
+            if prefix in attrs['atype']:
+                return attrs['atype']
+    raise IOError(f"Could not find GoVs with resid {resid} in chain {chain}.")
+
+def _in_region(resid, regions):
+    """
+    Check if resid falls in regions interval.
+    """
+    for low, up in regions:
+        if low <= resid <= up:
+            return True
+    return False
+
+class GetGo():
+    """
+    Generate the Go model interaction parameters.
+    """
+    def __init__(self,
+                 contact_map,
+                 cutoff_short,
+                 cutoff_long,
+                 go_eps,
+                 res_dist,
+                 res_graph,
+                 regions=None,
+                 water_bias=None,
+                 idp_water_bias=None,
+                 prefix="VS"):
+
+        self.contact_map = contact_map
+        self.cutoff_short = cutoff_short
+        self.cutoff_long = cutoff_long
+        self.go_eps = go_eps
+        self.res_dist = res_dist
+        self.itp_file = open('GoVirtIncludes.itp', "w", encoding='UTF-8')
+        self.prefix = prefix
+        self.res_graph = res_graph
+        self.water_bias = water_bias
+        self.idp_water_bias = idp_water_bias
+        self.regions = regions
+
+    def write_go_structural_bias(self, molecule):
```

Comment:
This function does way more than writing

Reply:
yep it was based on the old implementation. Now it's completely reworked.